### PR TITLE
Update past foreign secretaries

### PR DESCRIPTION
**Pull request by chris-gds**
## What

Update [Past Foreign Secretaries Page](https://www.gov.uk/government/history/past-foreign-secretaries) to apply new Design + Design system + use Components

## Why

On-going work to align architecture, have pages benefit from Accessibility improvements

## Visuals

| Before | After |
| ----------- | ----------- |
| www gov uk_government_history_past-foreign-secretaries(iPhone-6_7_8-Plus) | www gov uk_government_history_past-foreign-secretaries(Moto G4) |
|  |
| www gov uk_government_history_past-foreign-secretaries | www gov uk_government_history_past-foreign-secretaries (1) | 

### Done when 

- [x] [Release merged](https://github.com/alphagov/whitehall/pull/6242)
- [x] Design Reviewed
- [ ] QA Content

### Grouped related PRs
https://github.com/alphagov/whitehall/pull/6236
https://github.com/alphagov/whitehall/pull/6243
https://github.com/alphagov/whitehall/pull/6241

## Anything else?

Add link to image card heading - this is not dependant
https://github.com/alphagov/govuk_publishing_components/pull/2277

[Implementing a semantic list](https://github.com/alphagov/whitehall/pull/6243/commits/992f5a1bb28dd8c20912c03f583803273bf3fcc8) in this context + the [Design System layout](https://design-system.service.gov.uk/styles/layout/) could be done various different ways.  This might prompt a change during review.

### CODE REVIEW

**Comment by jon-kirwan** [comment]
All looks good to me. Just one small comment to add with relation to -

https://github.com/alphagov/whitehall/pull/6243/files#diff-4c36183021f514ff3edc854264a1c1c1d3b5a954d499f383bd9a56cd5729cfeaR21

```
  .govuk-heading-s--foreign-secretary-profile {
    margin-bottom: 0;
  }
```
I wondered if an alternative approach could be to replace this CSS with the govuk-!-margin-bottom-0 class?

**Comment by chris-gds** [comment]
> I wondered if an alternative approach could be to replace this CSS with the govuk-!-margin-bottom-0 class?

Good call, should work the same, i'll update and remove those extra lines.

**Comment by chris-gds** [comment]
@jon-kirwan updated! Tests _were_ failing but needed a rebuild.

**Review by jon-kirwan — APPROVED** [review]

File: app/views/past_foreign_secretaries/index.html.erb [hunk 1]
```diff
@@ -12,736 +12,769 @@
...
+<div class="historic-people-index__section-row" role="list">
+  <div class="govuk-grid-row">
+    <div role="listitem" class="govuk-grid-column-one-quarter">
+      <%# Custom heading as links to external page %>
+      <h3 class="govuk-heading-s govuk-!-margin-bottom-0">
+        <a class="govuk-link" href="/government/people/dominic-raab/">Dominic Raab</a>
+      </h3>
+      <p class="govuk-body">2019 to present</p>
+    </div>
+    <div role="listitem" class="govuk-grid-column-one-quarter">
```
Comment (maxgds) [inline app/views/past_foreign_secretaries/index.html.erb:177]: This (and the other PRs) all seem very repetitive. I think there would be value in making hashes of the data then looping through them and using a single partial to populate with the data along the lines of this:

```
<h3 class="govuk-heading-s govuk-!-margin-bottom-0">
      <% if link.present? %>  
         <a class="govuk-link" href="*variable here*">*variable here*</a>
     <% else %>
          *variable here*
    <% end %>
      </h3>
<div role="listitem" class="govuk-grid-column-one-quarter">
      <%= render "govuk_publishing_components/components/heading", {
        text: *variable here*,
        heading_level: 3,
        font_size: "s",
      } %>
      <p class="govuk-body">*variable here*</p>
    </div>
```
You could potentially expand it out to the image card examples as well? I don't think it's common enough to be worth making a component, but it's not at all DRY to be seeing repeated HTML markup like this.

**Review by maxgds — CHANGES_REQUESTED** [review]
While the output is sound, I have reservations around the div vs li markup and am concerned about the level of repetition that exists. I think the 3 pages dealt with in the one trello ticket could be sharing more code because of their similarities and would like you to explore that a bit more rather than merging as-is.

File: app/views/past_foreign_secretaries/index.html.erb [hunk 2]
```diff
@@ -12,736 +12,769 @@
...
+    <%= render "govuk_publishing_components/components/heading", {
+      text: "21st century",
+      padding: true,
+      border_top: 2,
+      font_size: "l"
+    } %>
+    <hr class="govuk-section-break govuk-section-break--m govuk-section-break--visible">
+  </div>
+</div>
+<div class="historic-people-index__section-row" role="list">
```
[the remark this answers is not in the document]
Comment (chris-gds) [reply app/views/past_foreign_secretaries/index.html.erb:168]: Yeah it's the same problem across 3 - if I markup as a list I lose the Design system, if I use the Design System I lose the markup but replicate with aria.  Once a consensus is reached i'll update across all

File: app/views/past_foreign_secretaries/index.html.erb (hunk 1 above)
Comment (chris-gds) [reply app/views/past_foreign_secretaries/index.html.erb:177]: As this is hardcode - do you have a good example to ref to use for this context, is this worth adding to `content_item` or locale .[yml](https://github.com/alphagov/whitehall/blob/175fec283eab3efa5e61698d4eb90cf47cefe869/config/locales/en.yml) file?

Comment (maxgds) [reply app/views/past_foreign_secretaries/index.html.erb:177]: I don't think it fits in a locale file as the strings are not really translatable (names/dates) and they're definitely not page furniture and content items are complicated. I'd keep it in situ as arrays of hashes like: 
```
C21 = [
{name:"Boris Johnson", dates: "2020-2021"}, 
{name:"Theresa May", dates: "xxxx-xxxx", link: "/government/whatever"}, 
]

C21.each do
	call partial
end```
Excuse any bad syntax, but you get the drift?

Comment (chris-gds) [reply app/views/past_foreign_secretaries/index.html.erb]: So! This has been split out, added to the controller and the partial iterates over.

File: app/assets/stylesheets/frontend/views/_history-people.scss [hunk 3]
```diff
@@ -1,68 +1,37 @@
...
+.historic-people-index__section-row-header {
+  @include govuk-media-query($from: tablet) {
+    margin-top: govuk-spacing(2);
+    margin-bottom: govuk-spacing(4);
   }
 
-  .featured-profiles {
-    margin-bottom: $gutter;
+  // unique image card styles for "past-foreign-secretaries" & "past-past-chancellor"
+  .gem-c-image-card__list-item-link {
```
Comment (maxgds) [inline app/assets/stylesheets/frontend/views/_history-people.scss:8]: This doesn't feel great, overriding component styles. Willing to let it pass if we absolutely have to but I'd suggest considering either/or 1) Checking with design that the default isn't good enough, 2) making this style the default for the "unlinked links".

**Review by maxgds — CHANGES_REQUESTED** [review]
This is looking much better from a DRY perspective, but I think I haven't communicated very well what I was aiming for. The three pages on this one trello card are using the same patterns throughout, there seems to be value in sharing not just the css, but the partial here. I  think this is a good first step towards uniting them like that, but if that's the end game the person and image partial names and locations should be more generic. I know the underlying code that builds each page differs but do you think this is achievable?

File: app/assets/stylesheets/frontend/views/_history-people.scss (hunk 3 above)
Comment (andysellick) [reply app/assets/stylesheets/frontend/views/_history-people.scss:8]: We shouldn't be overriding component styles in applications - it breaks component isolation - e.g. if we were to change the name of this class in the component, this override would suddenly stop working and we wouldn't know about it.

If the link styles definitely need to be different in this one situation, please add it as an option to the component. Do they definitely need to be different?

Comment (chris-gds) [reply app/assets/stylesheets/frontend/views/_history-people.scss:8]: Agreed. 🤔  
Let me do the reverse and reduce the font everywhere.

**Comment by chris-gds** [comment]
Moving into Draft (aim to close) as combined PR raised: https://github.com/alphagov/whitehall/pull/6264

**Pull request closed without merging**